Src/Support/Google.Apis.Auth/GoogleJsonWebSignature.cs

```diff
@@ -14,15 +14,191 @@
+                throw new InvalidJwtException($"Java Web Token exceeds maximum allowed length of {MaxJwtLength}");
+            }
+            var parts = jwt.Split('.');
+            if (parts.Length != 3)
+            {
+                throw new InvalidJwtException($"Java Web Token must consist of Header, Payload, and Signature");
+            }
+            
+            // Decode the three parts of the JWT: header.payload.signature
+            Header header = NewtonsoftJsonSerializer.Instance.Deserialize<Header>(Base64UrlToJson(parts[0]));
+            Payload payload = NewtonsoftJsonSerializer.Instance.Deserialize<Payload>(Base64UrlToJson(parts[1]));
+            byte[] signature = Base64UrlDecode(parts[2]);
+
+            // Verify algorithm in JWT
+            if (header.Algorithm != "RS256")
+            {
+                throw new InvalidJwtException("JWT algorithm must be 'RS256;");
+            }
+
+            // Verify signature
+            byte[] hash;
+            using (var hashAlg = SHA256.Create())
+            {
+                hash = hashAlg.ComputeHash(Encoding.ASCII.GetBytes($"{parts[0]}.{parts[1]}"));
+            }
+            bool verifiedOk = false;
+            foreach (var googleCert in await GetGoogleCertsAsync(clock, forceGoogleCertRefresh, certsJson))
+            {
+#if NET45
+                verifiedOk = ((RSACryptoServiceProvider)googleCert).VerifyHash(hash, Sha256Oid, signature);
+#elif NETSTANDARD1_3
+                verifiedOk = googleCert.VerifyHash(hash, signature, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+#else
+#error Unsupported platform
+#endif
+                if (verifiedOk)
+                {
+                    break;
+                }
+            }
+            if (!verifiedOk)
+            {
+                throw new InvalidJwtException("JWT invalid: unable to verify signature.");
+            }
+
+            // Verify iss, iat and exp claims
+            if (payload.Issuer != "https://accounts.google.com" && payload.Issuer != "accounts.google.com")
+            {
+                throw new InvalidJwtException("JWT issuer incorrect. Must be 'https://accounts.google.com' or 'accounts.google.com'");
+            }
+            if (payload.IssuedAtTimeSeconds == null || payload.ExpirationTimeSeconds == null)
+            {
+                throw new InvalidJwtException("JWT must contain 'iat' and 'exp' claims");
+            }
+            var now = (clock.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds;
+            if (now < payload.IssuedAtTimeSeconds.Value || now > payload.ExpirationTimeSeconds.Value)
+            {
+                throw new InvalidJwtException("JWT is not yet valid, or has expired.");
+            }
+
+            // All verification passed, return payload.
+            return payload;
+        }
+
+        private static string Base64UrlToJson(string base64Url) => Encoding.UTF8.GetString(Base64UrlDecode(base64Url));
```

Comment:
This method name is a bit confusing, I think that it should be `Base64UrlToString()` instead, since that is what it is doing. Whether the string is JSON or not is a different matter.